pkg/monitor/agent/agent.go

```diff
@@ -78,12 +80,21 @@ type Agent struct {
 // goroutine and close all registered listeners.
 // Note that the perf buffer reader is started only when listeners are
 // connected.
-func NewAgent(ctx context.Context) *Agent {
+func NewAgent(ctx context.Context, optionalLimit ...int) *Agent {
```

Comment:
I think the type signature doesn't really match the semantics, and there aren't _that_ many calls to `NewAgent` to change. If we really wanted to make this optional, the options pattern would work, but in this case I'd just opt for always passing the limit, the zero value is the default anyway.

Reply:
Done